### PR TITLE
[EAGLE-4938] Added Runners's model upload examples

### PR DESCRIPTION
**Pull request by luv-bansal**
Added Runners's model upload examples of different categories. 
- LLM
- vLLM LLM model
- Image Classification
- Image Detection
- Speech Recognition (audio-to-text)
- Model for testing purpose (python_string_cat)

Note: 
Before Model Upload examples are at [examples/model_upload](https://github.com/Clarifai/examples/tree/main/model_upload) but I think it should be at `examples/models/model_upload`

### CODE REVIEW

**Comment by zeiler** [comment]
Good call on the new location as we have model_eval and mod
el_train there too.

File: models/model_upload/llms/openai-gpt4/config.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  id: "model_id"
+  user_id: "user_id"
+  app_id: "app_id"
+  model_type_id: "text-to-text"
+
+build_info:
+  python_version: "3.10"
+
+inference_compute_info:
+  cpu_limit: "2"
```
Comment (zeiler) [inline models/model_upload/llms/openai-gpt4/config.yaml:13]: this is likely fine with 1 core or 500m for half a core.

File: models/model_upload/llms/llama-3-8b-instruct/config.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  user_id: "user_id"
+  app_id: "app_id"
+  model_type_id: "text-to-text"
+
+build_info:
+  python_version: "3.11"
+
+inference_compute_info:
+  cpu_limit: "2"
+  cpu_memory: "13Gi"
```
Comment (zeiler) [inline models/model_upload/llms/llama-3-8b-instruct/config.yaml:14]: same

File: models/model_upload/llms/vllm-mistral-7b-instruct/config.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  user_id: "user_id"
+  app_id: "app_id"
+  model_type_id: "text-to-text"
+
+build_info:
+  python_version: "3.11"
+
+inference_compute_info:
+  cpu_limit: "2"
+  cpu_memory: "13Gi"
```
Comment (zeiler) [inline models/model_upload/llms/vllm-mistral-7b-instruct/config.yaml:14]: same

File: models/model_upload/speech-recognition/openai-whisper/config.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  id: "model_id"
+  user_id: "user_id"
+  app_id: "app_id"
+  model_type_id: "audio-to-text"
+
+build_info:
+  python_version: "3.10"
+
+inference_compute_info:
+  cpu_limit: "2"
```
Comment (zeiler) [inline models/model_upload/speech-recognition/openai-whisper/config.yaml:13]: likely fine with 1 or 500m cores

File: models/model_upload/speech-recognition/openai-whisper/config.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  cpu_memory: "2Gi"
```
Comment (zeiler) [inline models/model_upload/speech-recognition/openai-whisper/config.yaml:14]: wonder if memory for this and the other openAI wrapper can be lower too?

File: models/model_upload/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ - `requirements.txt` - model requirements python libraries
+ - `model.py` - the implementation of the model.
+ - `config.yaml` - contains compute resource information and other details needed to build the Docker image and upload the model to the to Clarifai Platform.
+
+### Model Upload Process
+- Define a appropriate model id, and your user_id and app_id in the config.yaml of model example for where the model should be uploaded to Clarifia Platform
+
+- Edit the model.py to do as you’d like for the implementation of your model
+
+- Now that your custom model is ready, you can serve the model in production
```
[the remark this answers is not in the document]
Comment (luv-bansal) [reply models/model_upload/README.md]: I have updated the README file, which now contains much more information about Model Upload. Let me know if anything else needs to be added or modified in the README file.

[the remark this answers is not in the document]
Comment (luv-bansal) [reply models/model_upload/README.md]: I completely agree with this. I was literally thinking the same thing. Until we have a proper CLI, we can create a script at `clarifai-python/clarifai/runners/models/model_test_locally.py`, allowing users to test models locally by running `python -m clarifai.runners.models.model_test_locally --model_path <model_path>.`
> We could add some smoke tests to the SDK CLI in the future to like pip install the requirements, download checkpoints, then based on the type of the model send it some dummy data to make sure it works with the local server. That automation for the user would be great

**Review by zeiler — APPROVED** [review]

**Pull request merged**